clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -537,6 +536,7 @@ public void handle(SyncGroupResponse syncResponse,
             if (error == Errors.NONE) {
                 sensors.syncLatency.record(response.requestLatencyMs());
                 future.complete(syncResponse.memberAssignment());
+                AbstractCoordinator.this.rejoinNeeded = false;
```

Comment:
@dguy @hachikuji if it sounds good to you I can go ahead and make this change while merging.